tools/pre-commit.sh

```diff
@@ -0,0 +1,10 @@
+#!/bin/sh
+# Copyright 2018 the Deno authors. All rights reserved. MIT license.
+# The script for git pre-commit hook.
+# This script formats the source code and adds them back to staging.
+staged=$(git diff --cached --name-only --diff-filter=ACM | tr '\n' ' ')
+[ -z "$staged" ] && exit 0
+echo Formatting files
+./tools/format.py
+echo "$staged" | xargs git add
```

Comment:
I don't want it to call "git add", because I very often don't commit all changes.

For this to be acceptable you need to run the contents from the index through the formatter, not the work tree file.

Reply:
This script formats all files, but it only adds staged files (because `$staged` is created from `git diff --cached ...` which only includes staged files.)

Reply:
I checked that this doesn't include unstaged changes (at least on my mac).

Reply:
@kt3k what if you only add some lines i.e. `git add -p`

Reply:
What @hayd said is what I meant. 
I do that all the time.

Reply:
I didn't think of that case enough... I'll try to think about the solution.

BTW `lint-staged` authors seem trying to solve a similar problem. ref. https://github.com/okonet/lint-staged/pull/75
We might use the above tool when they shipped the above change?